src/AbstractCollection.php

```diff
@@ -57,9 +62,188 @@ public function remove($element): bool
         if (($position = array_search($element, $this->data, true)) !== false) {
             unset($this->data[$position]);
 
+
             return true;
         }
 
         return false;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function column(string $propertyOrMethod): array
+    {
+        $temp = [];
+
+        foreach ($this->data as $item) {
+            $temp[] = $this->extractValue($item, $propertyOrMethod);
+        }
+
+        return $temp;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function first()
+    {
+        if (empty($this->data)) {
+            throw new OutOfBoundsException('Can\'t determine first item. Collection is empty');
+        }
+
+        $temp = array_merge([], $this->data);
```

Comment:
What's the purpose of this line?

Reply:
Because of the nature of `array_shift` and `array_pop` which will alter the target array. This will make sure I will lose the reference of the current data array.

And because the key of the data list is not necessary numeric I just use internal function to get the last or first item.